numpy/lib/stride_tricks.py

```diff
@@ -194,7 +194,7 @@ def _broadcast_shape(*args):
         # ironically, np.broadcast does not properly handle np.broadcast
         # objects (it treats them as scalars)
         # use broadcasting to avoid allocating the full array
-        b = broadcast_to(0, b.shape)
+        b = broadcast_to.__skip_array_function__(0, b.shape)
```

Comment:
All good, except this one (maybe!). This gets called from inside the `np.vectorize` class, which is not hidden away behind `__array_function__`. So I *think* this may be wrong?

Reply:
`vectorize` always coerces with `asarray()` or `asanyarray()`, so I think this is fine, as a (minor) optimization.

Reply:
Are these optimizations really worth making?